sdk/src/main/java/com/qonversion/android/sdk/dto/QPurchaseOptions.kt

```diff
@@ -11,33 +18,82 @@ class QPurchaseOptions internal constructor (
     internal val oldProduct: QProduct? = null,
     internal val updatePolicy: QPurchaseUpdatePolicy? = null
 ) {
+    /**
+     * The builder of QPurchaseOptions instance.
+     *
+     * This class contains a variety of methods to customize the purchase behavior.
+     * You can call them sequentially and call [build] finally to get the [QPurchaseOptions] instance.
+     */
     class Builder {
         private var contextKeys: List<String>? = null
         private var offerId: String? = null
         private var applyOffer: Boolean = true
         private var oldProduct: QProduct? = null
         private var updatePolicy: QPurchaseUpdatePolicy? = null
 
+        /**
+         * Set the context keys associated with a purchase.
+         *
+         * @param contextKeys context keys for the purchase.
+         * @return builder instance for chain calls.
+         */
         fun setContextKeys(contextKeys: List<String>): QPurchaseOptions.Builder = apply {
             this.contextKeys = contextKeys
         }
 
+        /**
+         * Set the offer Id to the purchase.
+         * If [offerId] is not specified, then the default offer will be applied. To know how we choose
+         * the default offer, see [QProductStoreDetails.defaultSubscriptionOfferDetails].
+         * @param offerId context keys for the purchase.
+         * @return builder instance for chain calls.
+         */
         fun setOfferId(offerId: String): QPurchaseOptions.Builder = apply {
             this.offerId = offerId
         }
 
+        /**
+         * Set context keys associated with a purchase.
+         *
+         * @param oldProduct context keys for the purchase.
+         * @return builder instance for chain calls.
+         */
         fun setOldProduct(oldProduct: QProduct): QPurchaseOptions.Builder = apply {
             this.oldProduct = oldProduct
         }
 
+        /**
+         * Set the update policy for the purchase.
+         * If the [updatePolicy] is not provided, then default one
+         * will be selected - [QPurchaseUpdatePolicy.WithTimeProration].
+         * @param updatePolicy update policy for the purchase.
+         * @return builder instance for chain calls.
+         */
         fun setUpdatePolicy(updatePolicy: QPurchaseUpdatePolicy): QPurchaseOptions.Builder = apply {
             this.updatePolicy = updatePolicy
         }
 
-        fun removeOffer() = apply {
+        /**
+         * Call this function to remove any intro/trial offer from the purchase (use only a bare base plan).
+         * @return builder instance for chain calls.
+         */
+        fun removeOffer(): QPurchaseOptions.Builder = apply {
             this.applyOffer = false
         }
 
+        /**
+         * Set offer for the purchase.
+         * @param offer concrete offer which you'd like to purchase.
+         * @return builder instance for chain calls.
+         */
+        fun setOffer(offer: QProductOfferDetails): QPurchaseOptions.Builder = apply {
```

Comment:
let's place all these thee methods near each other, in order `setOffer`, `setOfferId`, `removeOffer`